knimin/handlers/barcode_util.py

```diff
@@ -182,28 +63,54 @@ def get_ag_details(self, barcode):
             ag_details['email_type'] = "-1"
         return div_id, message, ag_details
 
-    def update_ag_barcode(self, barcode):
-        msg2 = msg3 = None
-        sent_date = self.get_argument('sent_date', None)
-        general_name = "American Gut participant"
-        login_user = self.get_argument('login_user', None)  # kit owner name
-        if login_user == "None":
-            login_user = general_name
-        send_mail = self.get_argument('send_mail', None)
-        if send_mail is not None:
-            email_type = self.get_argument('email_type', None)
-            subject = body_message = ""
-            sample_time = self.get_argument('sample_time', None)
-            sample_date = self.get_argument('sample_date', None)
-            if email_type == '0':
-                subject = ('Follow up on Your American Gut Sample with '
-                           'Barcode %s' % barcode)
-                body_message = """
+    def update_ag_barcode(self, barcode, login_user, login_email, email_type,
+                          sent_date, send_mail, sample_date, sample_time,
+                          other_text):
+        email_msg = ag_update_msg = None
+        if all([send_mail is not None, login_email is not None,
+               login_email != '']):
+            subject, body_message = self._build_email(
+                login_user, barcode, email_type, sample_date, sample_time)
+            if body_message != '':
+                try:
+                    send_email(body_message, subject, login_email)
+                    sent_date = time.now()
+                    email_msg = ("Sent email successfully to kit owner %s" %
+                                 login_email)
+                except:
+                    email_msg = ("Email sending to (%s) failed (barcode: %s)!"
+                                 "<br/>" % (login_email, barcode))
+        sample_issue = self.get_argument('sample_issue', [])
+        moldy = overloaded = other = 'N'
+        if 'moldy' in sample_issue:
+            moldy = 'Y'
+        if 'overloaded' in sample_issue:
+            overloaded = 'Y'
+        if 'other' in sample_issue:
+            other = 'Y'
+        try:
+            db.updateAKB(barcode, moldy, overloaded, other, other_text,
+                         sent_date)
+            ag_update_msg = ("Barcode %s AG info was sucessfully updated" %
```

Comment:
minor, but on the principle of doing the least necessary within a scope, a better strategy here would be:

```python
ag_update_msg = ("Barcode %s AG update failed!!!" % barcode)
try:
    db.updateAKB(barcode, moldy, overloaded, other, other_text,
                         sent_date)
except:
    ag_update_msg = ("Barcode %s AG info was sucessfully updated" % barcode)
```